core/src/main/java/com/linecorp/armeria/server/metric/PrometheusVersion1ExpositionService.java

```diff
@@ -0,0 +1,119 @@
+import io.prometheus.metrics.expositionformats.ExpositionFormats;
+import io.prometheus.metrics.model.registry.PrometheusRegistry;
+
+/**
+ * Exposes Prometheus metrics in <a href="https://prometheus.io/docs/instrumenting/exposition_formats/">
+ * EXPOSITION FORMATS</a>.
+ */
+public final class PrometheusVersion1ExpositionService extends AbstractHttpService
+        implements TransientHttpService {
+
+    /**
+     * Returns a new {@link PrometheusVersion1ExpositionService} that exposes Prometheus metrics from
+     * {@link PrometheusRegistry#defaultRegistry}.
+     */
+    public static PrometheusVersion1ExpositionService of() {
+        return of(PrometheusRegistry.defaultRegistry);
+    }
+
+    /**
+     * Returns a new {@link PrometheusVersion1ExpositionService} that exposes Prometheus metrics from
+     * the specified {@link PrometheusRegistry}.
+     */
+    public static PrometheusVersion1ExpositionService of(PrometheusRegistry prometheusRegistry) {
+        return builder(prometheusRegistry).build();
+    }
+
+    /**
+     * Returns a new {@link PrometheusVersion1ExpositionServiceBuilder} created with
+     * {@link PrometheusRegistry#defaultRegistry}.
+     */
+    public static PrometheusVersion1ExpositionServiceBuilder builder() {
+        return builder(PrometheusRegistry.defaultRegistry);
+    }
+
+    /**
+     * Returns a new {@link PrometheusVersion1ExpositionServiceBuilder} created with the specified
+     * {@link PrometheusRegistry}.
+     */
+    public static PrometheusVersion1ExpositionServiceBuilder builder(PrometheusRegistry prometheusRegistry) {
+        return new PrometheusVersion1ExpositionServiceBuilder(
+                requireNonNull(prometheusRegistry, "prometheusRegistry"));
+    }
+
+    private final PrometheusRegistry prometheusRegistry;
+    private final ExpositionFormats expositionFormats = ExpositionFormats.init();
+    private final Set<TransientServiceOption> transientServiceOptions;
+
+    PrometheusVersion1ExpositionService(PrometheusRegistry prometheusRegistry,
+                                        Set<TransientServiceOption> transientServiceOptions) {
+        this.prometheusRegistry = prometheusRegistry;
+        this.transientServiceOptions = ImmutableSet.copyOf(transientServiceOptions);
+    }
+
+    @Override
+    protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+        final String accept = req.headers().get(HttpHeaderNames.ACCEPT);
+        final ByteBuf buffer = ctx.alloc().buffer();
+        final String format;
+        boolean success = false;
+        try (ByteBufOutputStream byteBufOutputStream = new ByteBufOutputStream(buffer)) {
+            final ExpositionFormatWriter writer = expositionFormats.findWriter(accept);
+            format = writer.getContentType();
+            writer.write(byteBufOutputStream, prometheusRegistry.scrape());
+            success = true;
+        } finally {
```

Comment:
What do you think of consolidating this class with the original one? This will help users perform a more smooth migration.
```java
class PrometheusExpositionService {
    @Deprecate
    public static PrometheusExpositionService builder(CollectorRegistry collectorRegistry) {
        ...
    }

    @Deprecate
    public static PrometheusExpositionServiceBuilder builder(CollectorRegistry collectorRegistry) {
        ...
    }

    public static PrometheusExpositionServiceBuilder builder(PrometheusRegistry prometheusRegistry) {

}
```

Reply:
> What do you think of consolidating this class with the original one? 

If so, users have to add two different modules even though they want the new one or the old one.
That's why I have created a separate class.

Reply:
On second thought, it would be clearer to split `PrometheusExpositionService` and `PrometheusMeterRegistries` as a separate module such as `:prometheus1`. This direction is an Armeria way to support multiple versions for a library like other dependencies.

Reply:
Alright, so you prefer @jrhee17's opinion: https://github.com/line/armeria/pull/5698#discussion_r1616507246